src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -84,18 +81,9 @@ public class RequestExecutor : IDisposable
 
         internal string LastServerVersion { get; private set; }
 
-        private HttpClient _httpClient;
-
         public HttpClient HttpClient
         {
-            get
-            {
-                var httpClient = _httpClient;
-                if (httpClient != null)
-                    return httpClient;
-
-                return _httpClient = GetHttpClient();
-            }
+            get => HttpClientFactory.GetHttpClient(_httpClientCacheKey, Conventions.CreateHttpClient);
```

Comment:
I wonder whether you could return same flag / boolean together with HttpClient that would say if the returned instance can be cached locally in `_httpClient` field. Depending if that shows up in the performance profiling now.

Reply:
Alternatively the factory itself could have a property indicating that.